R/custom_mosaic.R

```diff
@@ -0,0 +1,215 @@
+# minimal implementation of ggplot2 mosaic after ggmosaic was archived in CRAN
+#
+# This was heavily inspired by github.com/haleyjeppson/ggmosaic package but
+# simplified to only support 2 categorical variables
+
+#' Mosaic Rectangles Layer for ggplot2
+#'
+#' Adds a mosaic-style rectangles layer to a ggplot, visualizing the
+#' joint distribution of categorical variables.
+#' Each rectangle's size reflects the proportion of observations for
+#' combinations of `x` and `fill`.
+#'
+#' @param mapping Set of aesthetic mappings created by `aes()`. Must specify `x` and `fill`.
+#' @param data The data to be displayed in this layer.
+#' @param stat The statistical transformation to use on the data. Defaults to `"rects"`.
+#' @param position Position adjustment. Defaults to `"identity"`.
+#' @param ... Other arguments passed to `layer()`.
+#' @param na.rm Logical. Should missing values be removed?
+#' @param show.legend Logical. Should this layer be included in the legends?
+#' @param inherit.aes Logical. If `FALSE`, overrides default aesthetics.
+#'
+#' @return A ggplot2 layer that adds mosaic rectangles to the plot.
+#'
+#' @examples
+#' df <- data.frame(RACE = c("Black", "White", "Black", "Asian"), SEX = c("M", "M", "F", "F"))
+#' library(ggplot2)
+#' ggplot(df, aes(x = RACE, fill = SEX)) +
+#'   geom_rects()
```

Comment:
Typo that is making the CI to fail.

```suggestion
#'   geom_rect()
```

However, I wonder if this should be `geom_mosaic` instead:

```suggestion
#'   geom_mosaic()
```